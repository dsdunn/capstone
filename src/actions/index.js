export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const setProfile = profile => ({
  type: 'SET_PROFILE',
  profile
})

export const addCollections = (collections) => ({
  type: 'ADD_COLLECTIONS',
  collections
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})

export const updateCollectionsList = (collectionsList) => {
  type: 'UPDATE_COLLECTIONS_LIST',
  collectionsList
}