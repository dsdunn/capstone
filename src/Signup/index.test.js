import React from 'react';
import { shallow } from 'enzyme';
import { signIn } from '../actions'
import { Signup, mapDispatchToProps } from './';

describe('Signup', () => {
  let wrapper;
  let history;

  beforeEach(() => {
    history = { location: 'location',
                goBack: jest.fn(),
                push: jest.fn() }
    wrapper = shallow(<Signup history={ history }/>)
  })

  it('should start with default state of username, email, password1, password2, and error', () => {
    const actual = wrapper.state()
    const expected = {
      email: '',
      username: '',
      password1: '',
      password2: '',
      error: null
    }

    expect(actual).toEqual(expected)
  })

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: {id: 'email', value: 'sleepy' }}

    wrapper.instance().handleChange(mockEvent)

    const actual = wrapper.state('email')

    expect(actual).toEqual('sleepy')
  })

  it('should a user a user when createUser is called', () => {
    const mockResponse = { user: { uid: 666 }}

    wrapper.setState({ username: 'Billy'})
    const actual = wrapper.instance().createUser(mockResponse)

    const expected = { uid: 666, username: 'Billy'}

    expect(actual).toEqual(expected)
  })

  it('should reset state when resetForm is called', () => {
    wrapper.setState({ username: 'Billy'})
    wrapper.instance().resetForm()
    const actual = wrapper.state()
    const expected = {
      email: '',
      username: '',
      password1: '',
      password2: '',
      error: null
    }

    expect(actual).toEqual(expected)
  })

  it('should submit the form when handleSubmit is called', () => {
    const mockEvent = { preventDefault: jest.fn() }
    wrapper.instance().handleSubmit(mockEvent)
    const actual = wrapper.state('error')
    const expected = "Uh oh! make sure you fill out the required fields ...thats all of them"

    expect(actual).toEqual(expected)
    // expect(history.push).toHaveBeenCalled()
  })

  it('should validate the form when validate is called', () => {
    wrapper.setState({ password1: '123456', password2: '1234567' })
    const actual = wrapper.instance().validate()
    const expected = false

    expect(actual).toEqual(expected)
  })

  it('should validate the form when validate is called', () => {
    wrapper.setState({ password1: '123456',
                       password2: '123456',
                       email: 'heck@yeah.com',
                       username: 'Sparkle' })
    const actual = wrapper.instance().validate()
    const expected = true

    expect(actual).toEqual(expected)
  })

  it('should go call goBack when goBack', () => {
    wrapper.instance().goBack()
    expect(history.goBack).toHaveBeenCalled()
  })

  it('should return a props object with signIn', () => {
    const mockSignIn = jest.fn();
    const mockDispatch = jest.fn();
    const mockUser = { password1: '123456',
                       password2: '123456',
                       email: 'heck@yeah.com',
                       username: 'Sparkle' }

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = signIn(mockUser);

    mappedProps.signIn(mockUser);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})






















