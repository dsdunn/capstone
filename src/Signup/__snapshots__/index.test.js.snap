// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup
    history={
        Object {
            "goBack": [Function],
            "location": "location",
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="signup-close-link"
          exact="true"
          onClick={[Function]}
>
          <img
                    className="close-btn"
                    src="close.svg"
          />
</a>,
        <h5>
          Sign Up
</h5>,
        <label
          htmlFor="email"
>
          email
</label>,
        <input
          id="email"
          onChange={[Function]}
          type="email"
          value=""
/>,
        <label
          htmlFor="username"
>
          username
</label>,
        <input
          id="username"
          onChange={[Function]}
          value=""
/>,
        <label
          htmlFor="password1"
>
          password
</label>,
        <input
          id="password1"
          onChange={[Function]}
          type="password"
          value=""
/>,
        <label
          htmlFor="password2"
>
          confirm password
</label>,
        <input
          id="password2"
          onChange={[Function]}
          type="password"
          value=""
/>,
        <button>
          Save
</button>,
        <p
          className="signup-err-message"
/>,
      ],
      "className": "signup",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            className="close-btn"
            src="close.svg"
/>,
          "className": "signup-close-link",
          "exact": "true",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "close-btn",
            "src": "close.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "email",
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": "email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "email",
          "onChange": [Function],
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "username",
          "htmlFor": "username",
        },
        "ref": null,
        "rendered": "username",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "username",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "password",
          "htmlFor": "password1",
        },
        "ref": null,
        "rendered": "password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password1",
          "onChange": [Function],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "confirm password",
          "htmlFor": "password2",
        },
        "ref": null,
        "rendered": "confirm password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password2",
          "onChange": [Function],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save",
        },
        "ref": null,
        "rendered": "Save",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "signup-err-message",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="signup-close-link"
            exact="true"
            onClick={[Function]}
>
            <img
                        className="close-btn"
                        src="close.svg"
            />
</a>,
          <h5>
            Sign Up
</h5>,
          <label
            htmlFor="email"
>
            email
</label>,
          <input
            id="email"
            onChange={[Function]}
            type="email"
            value=""
/>,
          <label
            htmlFor="username"
>
            username
</label>,
          <input
            id="username"
            onChange={[Function]}
            value=""
/>,
          <label
            htmlFor="password1"
>
            password
</label>,
          <input
            id="password1"
            onChange={[Function]}
            type="password"
            value=""
/>,
          <label
            htmlFor="password2"
>
            confirm password
</label>,
          <input
            id="password2"
            onChange={[Function]}
            type="password"
            value=""
/>,
          <button>
            Save
</button>,
          <p
            className="signup-err-message"
/>,
        ],
        "className": "signup",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="close-btn"
              src="close.svg"
/>,
            "className": "signup-close-link",
            "exact": "true",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "close-btn",
              "src": "close.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "email",
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": "email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "onChange": [Function],
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "username",
            "htmlFor": "username",
          },
          "ref": null,
          "rendered": "username",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "username",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "password",
            "htmlFor": "password1",
          },
          "ref": null,
          "rendered": "password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password1",
            "onChange": [Function],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "confirm password",
            "htmlFor": "password2",
          },
          "ref": null,
          "rendered": "confirm password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password2",
            "onChange": [Function],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "signup-err-message",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
